governator-core/src/main/java/com/netflix/governator/ManagedInstanceAction.java

```diff
@@ -0,0 +1,39 @@
+package com.netflix.governator;
+
+import java.lang.ref.Reference;
+import java.util.concurrent.Callable;
+
+/**
+ * Runnable that applies one or more LifecycleActions to a managed instance T or Reference<T>.
+ * For Reference<T> the action is invoked on a best-effort basis, if the referent is non-null
+ * at time run() is invoked
+ */
+public final class ManagedInstanceAction implements Callable<Void> {
```

Comment:
Runnable eats or wraps the exception in a RuntimeException; Callable.call is allowed to throw Exception 'as-is'.